react-components/jsx.md

```diff
@@ -106,36 +98,46 @@ In most contexts, we can wrap JavaScript expressions in parentheses `()`, which
 
 ### !end-callout
 
-### JSX Objects With Children: `App.js`
+### JSX Objects With Children: `App.jsx`
 
-Let's revisit this `src/App.js`:
+Let's revisit this `src/App.jsx`:
 
 ```js
 function App() {
+  const [count, setCount] = useState(0)
+
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
+    <>
+      <div>
+        <a href="https://vitejs.dev" target="_blank">
+          <img src={viteLogo} className="logo" alt="Vite logo" />
+        </a>
+        <a href="https://react.dev" target="_blank">
+          <img src={reactLogo} className="logo react" alt="React logo" />
+        </a>
+      </div>
+      <h1>Vite + React</h1>
+      <div className="card">
+        <button onClick={() => setCount((count) => count + 1)}>
+          count is {count}
+        </button>
         <p>
-          Edit <code>src/App.js</code> and save to reload.
+          Edit <code>src/App.jsx</code> and save to test HMR
```

Comment:
Should we define HMR somewhere? I bet we'll get a question. Apparently it's "hot module reloading" (i.e., the thing that automatically reloads the app when we edit the source).

Reply:
I can add a little blurb. Do you think it should be a callout or the expandable arrow?

I was thinking of the title being "HMR: Hot Module Reloading".

The explanation could be like: "Hot Module Reloading, or HMR, as mentioned in `App.jsx` is a feature that the Vite development server comes with that automatically reloads the app when we edit the source code."

Reply:
We could even tie it back to the point we made in the earlier lesson about turning off atuo save in VS Code. Something like how this is a really useful development feature, but because of how it can interact with editing files in VS Code, this is why we recommend turning off VS Code Auto Save.